bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/EntryLogTest.java

```diff
@@ -391,4 +489,57 @@ public void testGetEntryLogsSet() throws Exception {
 
         assertEquals(Sets.newHashSet(0L, 1L, 2L, 3L), logger.getEntryLogsSet());
     }
+
+    /**
+     * Fake Ticker to advance ticker as expected.
+     */
+    class FakeTicker extends Ticker {
+
+        private final AtomicLong nanos = new AtomicLong();
+
+        /** Advances the ticker value by {@code time} in {@code timeUnit}. */
+        public FakeTicker advance(long time, TimeUnit timeUnit) {
+            nanos.addAndGet(timeUnit.toNanos(time));
+            return this;
+        }
+
+        @Override
+        public long read() {
+            long value = nanos.getAndAdd(0);
+            return value;
+        }
+    }
+
+    /**
+     * Fake EntryLogger using customized cache to hide superclass'.
+     */
+    class FakeEntryLogger extends EntryLogger {
+        private Ticker ticker;
+        FakeEntryLogger(ServerConfiguration conf, LedgerDirsManager ledgerDirsManager,
+                        Ticker ticker) throws IOException{
+            super(conf, ledgerDirsManager);
+            this.ticker = ticker;
+        }
+
+        private final ThreadLocal<Cache<Long, EntryLogBufferedReadChannel>> logid2ReadChannel =
```

Comment:
Why are you overridding this? For the ticker? It would be better to always pass in a ticker, and have a #getTicker() method that you can override. This will cut down the amount of code you need to duplicate.

Reply:
Obviously, the EntryLogger#getTicker() should just return Ticker.systemTicker